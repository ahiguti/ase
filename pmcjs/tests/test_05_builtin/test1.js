
var core = Perl.Functions['CORE'];

var pvar = Perl.NewScalar("abcdef\n");
dump(pvar); print();
core.chomp(pvar);
dump(pvar); print();
pvar = Perl.NewScalar("abcdef\n");
core.chop(pvar);
dump(pvar); print();

print(core.chr(65));
print(core.crypt('foobar', 'aa'));
print(core.hex('ffff'));
print(core.index('abcdefabcdef', 'abc', 1));
print(core.index('abcdefabcdef', 'abc'));
print(core.lc('AbCdEf'));
print(core.lcfirst('AbCdEf'));
print(core.length('abcdef'));
print(core.length('あいうえお')); /* should be 5 */
print(core.oct('777'));
print(core.ord('A').toString(16)); /* 0x41 */
print(core.ord('あ').toString(16)); /* 0x3042 */
print(core.reverse('abcdef'));
print(core.rindex('abcdefabcdef', 'abc', 1));
print(core.rindex('abcdefabcdef', 'abc'));
print(core.sprintf('x=%d y=%d', 10, 20));
print(core.substr('abcdef', 2, 2));
print(core.substr('abcdef', 2));

var pvar = Perl.NewScalar("abcdef");
print(core.substr(pvar, 2, 2, "ABCD"));
print(pvar);

