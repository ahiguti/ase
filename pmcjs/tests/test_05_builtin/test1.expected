61 62 63 64 65 66 0a
61 62 63 64 65 66
61 62 63 64 65 66
A
aa9TsAg8bf3eo
65535
6
0
abcdef
abCdEf
6
5
511
41
3042
fedcba
0
6
x=10 y=20
cd
cdef
cd
abABCDef
